public/expressions/overlay_alerts.ts

```diff
@@ -0,0 +1,122 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+import { i18n } from '@osd/i18n';
+import {
+  ExpressionFunctionDefinition,
+  ExprVisLayers,
+} from '../../../../src/plugins/expressions/public';
+import { TimeRange, calculateBounds } from '../../../../src/plugins/data/common';
+import { VisLayers, VisLayerErrorTypes } from '../../../../src/plugins/vis_augmenter/public';
+import { convertAlertsToLayer, getAlerts, getMonitorName } from './helpers';
+
+type Input = ExprVisLayers;
+type Output = Promise<ExprVisLayers>;
+
+interface Arguments {
+  detectorId: string;
+}
+
+const name = 'overlay_alerts';
+
+export type OverlayAlertsExpressionFunctionDefinition = ExpressionFunctionDefinition<
+  'overlay_alerts',
+  Input,
+  Arguments,
+  Output
+>;
+
+
+export const overlayAlertsFunction = (): OverlayAlertsExpressionFunctionDefinition => ({
+  name,
+  type: 'vis_layers',
+  inputTypes: ['vis_layers'],
+  help: i18n.translate('data.functions.overlay_alerts.help', {
+    defaultMessage: 'Add an alert vis layer',
+  }),
+  args: {
+    monitorId: {
+      types: ['string'],
+      default: '""',
+      help: '',
+    },
+  },
+
+  async fn(input, args, context): Promise<ExprVisLayers> {
+    // Parsing all of the args & input
+    const monitorId = get(args, 'monitorId', '');
+    const timeRange = get(context, 'searchContext.timeRange', '') as TimeRange;
+    const origVisLayers = get(input, 'layers', {
+      layers: [] as VisLayers,
+    }) as VisLayers;
+    const parsedTimeRange = timeRange ? calculateBounds(timeRange) : null;
+    const startTimeInMillis = parsedTimeRange?.min?.unix()
+      ? parsedTimeRange?.min?.unix() * 1000
+      : undefined;
+    const endTimeInMillis = parsedTimeRange?.max?.unix()
+      ? parsedTimeRange?.max?.unix() * 1000
+      : undefined;
+
+    // If time range is invalid return the original vis layers.
+    if (startTimeInMillis === undefined || endTimeInMillis === undefined) {
+      console.log('start or end time invalid');
+      return {
+        type: 'vis_layers',
+        layers: origVisLayers,
+      };
+    }
+
+    const alerts = await getAlerts(monitorId, startTimeInMillis, endTimeInMillis);
+    let alertLayer;
+    const monitorName = await getMonitorName(monitorId);
+    if (monitorName === VisLayerErrorTypes.RESOURCE_DELETED) {
+      const error = {
+        type: 'RESOURCE_DELETED',
+        message: 'The monitor does not exist.'
+      }
+      alertLayer = convertAlertsToLayer(alerts, monitorId, monitorName, error);
+    } else if (monitorName === VisLayerErrorTypes.FETCH_FAILURE) {
+      const error = {
+        type: 'FETCH_FAILURE',
+        message: 'The user does not have permissions to the monitor.'
+      }
+      alertLayer = convertAlertsToLayer(alerts, monitorId, monitorName, error);
+    } else {
+      alertLayer = convertAlertsToLayer(alerts, monitorId, monitorName);
```

Comment:
Since this is common for all the `if...else..`, we should just set the error inside the `if..else..` and call this once with the set values.

Something like
```
let error: string | undefined;
if (monitorName === 'A') error = 'errorA'
else if (monitorName === 'B') error = 'errorB'

alertLayer = convertAlertsToLayer(alerts, monitorId, error);
```